common/djangoapps/status/tests.py

```diff
@@ -1,99 +1,58 @@
-        "edX/toy/2012_Fall" : "A toy story"
-        }"""
-
-    global_and_toy = """{
-        "global" : "Hello, Globe",
-        "edX/toy/2012_Fall" : "A toy story"
-        }"""
-
-    # json to use, expected results for course=None (e.g. homepage),
-    # for toy course, for full course.  Note that get_site_status_msg
-    # is supposed to return global message even if course=None.  The
-    # template just happens to not display it outside the courseware
-    # at the moment...
-    checks = [
-        (no_file, None, None, None),
-        (invalid_json, None, None, None),
-        (global_only, "Hello, Globe", "Hello, Globe", "Hello, Globe"),
-        (toy_only, None, "A toy story", None),
-        (global_and_toy, "Hello, Globe", "Hello, Globe<br>A toy story", "Hello, Globe"),
-    ]
-
     def setUp(self):
-        """
-        Fake course ids, since we don't have to have full django
-        settings (common tests run without the lms settings imported)
-        """
         super(TestStatus, self).setUp()
-        self.full_id = 'edX/full/2012_Fall'
-        self.toy_id = 'edX/toy/2012_Fall'
-        self.addCleanup(self.remove_status_file)
+        # Clear the cache between test runs.
+        cache.clear()
+        self.course_key = CourseLocator(org='TestOrg', course='TestCourse', run='TestRun')
+
+    @unittest.skipUnless(settings.ROOT_URLCONF == 'lms.urls', 'Test only valid in lms')
+    @ddt.data(
+        ("Test global message", "Test course message"),
+        (u" Ŧɇsŧ sŧȺŧᵾs", u"Ṫëṡẗ ċöüṛṡë ṡẗäẗüṡ "),
+        (u"", u"Ṫëṡẗ ċöüṛṡë ṡẗäẗüṡ "),
+        (u" Ŧɇsŧ sŧȺŧᵾs", u""),
+    )
+    @ddt.unpack
+    def test_get_site_status_msg(self, test_global_message, test_course_message):
+        """Test status messages in a variety of situations."""
 
-    def create_status_file(self, contents):
-        """
-        Write contents to settings.STATUS_MESSAGE_PATH.
-        """
-        with open(settings.STATUS_MESSAGE_PATH, 'w') as f:
-            f.write(contents)
+        # When we don't have any data set.
+        self.assertEqual(get_site_status_msg(None), None)
+        self.assertEqual(get_site_status_msg(self.course_key), None)
 
-    def clear_status_cache(self):
-        """
-        Remove the cached status message, if found
-        """
-        if cache.get('site_status_msg') is not None:
-            cache.delete('site_status_msg')
+        msg = GlobalStatusMessage.objects.create(message=test_global_message, enabled=True)
+        msg.save()
 
-    def remove_status_file(self):
-        """Delete the status file if it exists"""
-        if os.path.exists(settings.STATUS_MESSAGE_PATH):
-            os.remove(settings.STATUS_MESSAGE_PATH)
+        self.assertEqual(get_site_status_msg(None), test_global_message)
 
-    @ddt.data(*checks)
-    @ddt.unpack
-    def test_get_site_status_msg(self, json_str, exp_none, exp_toy, exp_full):
-        """run the tests"""
+        course_msg = CourseMessage.objects.create(
```

Comment:
What about the case in which there is no `CourseMessage`? (not just the empty-string)